### PR TITLE
Add initial CI workflow

### PR DESCRIPTION
**Pull request by zmughal**

### CODE REVIEW

File: t/basic.t [hunk 1]
```diff
@@ -12,19 +12,15 @@ diag join "\t", qw{Mod Lib};
...
 
-  run_ok(['gunzip', '-h'])
+  run_ok(['libdeflate-gunzip', '-h'])
     ->success
-    ->out_like(qr/^usage:\s+gunzip\s/mi);
-
-  run_ok(['checksum', '-h'])
-    ->success
-    ->out_like(qr/^usage:\s+checksum\s/mi);
+    ->out_like(qr/^usage:\s+libdeflate-gunzip(\.exe)?\s/mi);
```
Comment (kiwiroy) [inline t/basic.t:23]: Is this correct? Should it be `libdeflate-checksum`?

Comment (zmughal) [reply t/basic.t:23]: I’ve removed `checksum` completely in one of the commits.

Comment (zmughal) [reply t/basic.t:23]: It’s a test program: https://github.com/ebiggers/libdeflate/blob/v1.10/Makefile#L264

File: alienfile [hunk 2]
```diff
@@ -44,29 +47,52 @@ share {
...
 
+  my $CC = qq(CC=@{[ $Config::Config{cc} ]});
+  my $static_lib = 'libdeflate.a';
+  my $shared_lib = qq(libdeflate.@{[ $Config::Config{so} ]});
+  my @progs = map { "$_@{[ $Config::Config{_exe} ]}" } qw(gzip gunzip);
   build [
-    '%{gmake} libdeflate.a gzip gunzip checksum CFLAGS=-fPIC',
-    '%{gmake} libdeflate.so',
+    [ '%{gmake}',
+        $CC,
```
Comment (kiwiroy) [inline alienfile:56]: I guess this could be `'CC=%{cc}'` but since Config is loaded already.

File: alienfile [hunk 3]
```diff
@@ -44,29 +47,52 @@ share {
...
+    ],
     sub {
       my $build = shift;
       my @dirs;
       my %fileset = (
         (map { $_ => path('include')->child($_)->stringify } qw{libdeflate.h}),
-        (map { $_ => path('lib')->child($_)->stringify } qw{libdeflate.a libdeflate.so}),
-        (map { $_ => path('bin')->child($_)->stringify } qw{checksum gzip gunzip}),
+        (map { $_ => path('lib')->child($_)->stringify } ( $static_lib, $shared_lib )),
+        (map { $_ => path('bin')->child("libdeflate-$_")->stringify } @progs),
```
Comment (zmughal) [inline alienfile:71]: See <https://packages.debian.org/sid/amd64/libdeflate-tools/filelist> and <https://github.com/ebiggers/libdeflate/blob/v1.10/Makefile#L343-L346>.

File: alienfile (hunk 2 above)
Comment (zmughal) [reply alienfile:56]: Whichever you prefer! I can change it if you would like.

**Pull request merged**